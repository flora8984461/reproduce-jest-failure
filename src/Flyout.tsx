import React from "react";
import {
  Popover,
  PopoverTrigger,
  PopoverSurface,
  Button,
  PopoverProps,
} from "@fluentui/react-components";
import { QuestionCircle24Regular } from "@fluentui/react-icons";

type Props = Omit<PopoverProps, "children">;

export const Flyout: React.FC<Props> = (props) => {
  return (
    <Popover {...props}>
      <PopoverTrigger>
        <Button id="button">Open</Button>
      </PopoverTrigger>
      <PopoverSurface>
        <div>AAAA</div>
        <div>
          <QuestionCircle24Regular />
        </div>
      </PopoverSurface>
    </Popover>
  );
};
