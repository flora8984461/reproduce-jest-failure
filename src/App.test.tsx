import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Flyout } from "./Flyout";

describe("Component to test", () => {
  it("Render flyout", () => {
    const { container } = render(<Flyout open />);
    const content = screen.getByText("AAAA");
    expect(content).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
