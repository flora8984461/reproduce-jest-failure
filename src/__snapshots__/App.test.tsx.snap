// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component to test Render flyout 1`] = `
<div>
  <button
    aria-expanded="true"
    class="fui-Button rsawnvh"
    data-tabster="{\\"deloser\\":{}}"
    role="button"
    tabindex="0"
    type="button"
  >
    Open
  </button>
  <span
    hidden=""
  />
</div>
`;
